chat/client-api.md

```diff
@@ -0,0 +1,62 @@
+# Client API
+
+```typescript
+abstract class Client {
+  // ---------- Methods ----------------------------------------------- //
+
+  // initializes the client with persisted storage and a network connection
+  public abstract init(): Promise<void>;
+
+  // register a blockchain account with a public key / returns the public key
+  public abstract register(params: {
+    account: string;
+  }): Promise<string>;
+
```

Comment:
Currently we have only one register() method that registers at a keyserver
but we are missing a method for "registering" for private discoverability, that will create key pair, subscribes for private topic etc.

just a question if it is an MVP scope?

Reply:
That's a good point. Perhaps, we could add a method for "generatePublicKey()"?

Reply:
I think the method should return chat URI - similarly to `Sign.connect()`

 what about something like `registerPublicDiscoverability()` and `registerPrivateDiscoverability`

I think we could also introduce discoverability object, so user can opt in/out and track where is known as who.